Fix microform classification

awead opened the pull request:
Refactors MarcMediaTypeProcessor and removes its microfilm format logic in favor of using the translation map file.

Fixes #287

## Issues

Making the change to `formats_949t.properties` required changing the tests, which now seem to overly favor microfilm as a format. The provided marc fixture files seem to support this: they all look like actual microfilms. We might want to provide additional tests, with supporting fixtures, to ensure that the other formats are getting tested properly.

ruthtillman reviewed (commented):
I'll leave approval to Banu but from the process perspective of assigning media and format types, things look good. Happy to review once we've run an index.

awead commented:
This line jumped out at me: https://github.com/psu-libraries/psulib_traject/pull/290/files#diff-6fb6123bac6081673ad76669b5d71b46669f036707072682853683e2c0358c64R105

It seems to suggest that Microfilm is the "other" of the formats. It seems like we ought to have an example marc record that is truly `Other` so we have something to test against.

ruthtillman commented:
@awead while in their **current** status, (e.g. at Bindery etc) these two items are Other. https://catalog.libraries.psu.edu/?f%5Baccess_facet%5D%5B%5D=Other

so if we wanted to get a MARC record example for "Other" to put in our tests, we could download either of them.

banukutlu commented on lib/marc_media_type_processor.rb:
```diff
@@ -1,114 +1,64 @@
+# media types to records
 class MarcMediaTypeProcessor
-  def initialize
-    freeze
```
@awead fwiw we had seen concurrency issues so ended up addind freeze to make it stateless and thread-safe, related PR: https://github.com/psu-libraries/psulib_traject/pull/122/files. So, I just wondered about that you haven't seen such behavior?

awead replied:
Interesting! Yes, you would run into thread issues with JRuby. I think we can make it thread-safe and refactor at the same time. Let me update it.

banukutlu commented:
@awead Looks great to me! I agree, as you mentioned we need additional specs or just need to update to use other records for the current types that are now resolving as microfilm.

banukutlu commented:
> @ruthtillman @banukutlu I don't think we have to have additional marc files to test each of the different scenarios, we just have to test the logic. I kind of hinted at that here: https://github.com/psu-libraries/psulib_traject/pull/290/files#diff-6fb6123bac6081673ad76669b5d71b46669f036707072682853683e2c0358c64R107
> 
> You can fake a record with the requisite parts, or just mock it completely. I'll work something up and post it.

yup that works too 👍  another similar approach we used in other specs https://github.com/psu-libraries/psulib_traject/blob/main/spec/lib/traject/subjects_spec.rb#L15. Either way is fine, with fixture files specs look a bit less crowded but it does not matter which everway you like is fine 👌

banukutlu commented:
> @ruthtillman @banukutlu I don't think we have to have additional marc files to test each of the different scenarios, we just have to test the logic. I kind of hinted at that here: https://github.com/psu-libraries/psulib_traject/pull/290/files#diff-6fb6123bac6081673ad76669b5d71b46669f036707072682853683e2c0358c64R107
> 
> You can fake a record with the requisite parts, or just mock it completely. I'll work something up and post it.

oh yeah cannot test every different scenario/format etc, did not do that before either, as you said just testing the different logic makes sense..

awead commented:
@banukutlu Ok, I've added two more commits to refactor MarcFormatProcessor. I've removed some of the duplicated tests and added additional ones should should get the test coverage up to %100 for the processor.

banukutlu commented on spec/spec_helper.rb:
```diff
@@ -32,7 +32,18 @@
+end
+
 RSpec.configure do |config|
+  config.before(:all) do
```
this refactor is :tada: 👌  🎉   neat and cleaner!

banukutlu reviewed (commented):
thanks @awead!
